Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OpenVRDeviceManager.cs

```diff
@@ -179,35 +181,62 @@ private GenericOpenVRController GetOrAddController(XRNodeState xrNodeState)
                     break;
             }
 
-            IMixedRealityInputSource inputSource = InputSystem?.RequestNewGenericInputSource($"{CurrentControllerType} Controller {controllingHand}");
+            Type controllerType;
+
+            switch (CurrentControllerType)
```

Comment:
Seems crazy to have two separate identical switch statements, any change we can reorganize it to make it tidier?

Reply:
I wish we could but there's things that need to happen in a specific order.

1. The pointers need to know the controller type (hence the first switch)
2. The controllers need to request a new Input Source using the generated pointer array
3. We need to actually create the concrete controller reference (hence the second switch)\
4. Then call setup config on the new controller
5. Then set the new controller on each of the new pointers

Reply:
Unless you know of a way to create a concrete instance from a type reference

Reply:
Would `Activator.CreateInstance` not work for this?

Reply:
Good question. I'll try it out.

Reply:
@Jarodshow how would you pass the constructor parameters in this case?

Reply:
[CreateInstance(Type, Object[])](https://msdn.microsoft.com/en-us/library/wcxyzt4d(v=vs.110).aspx), Examples are on the page :)